packages/binding-mqtt/src/mqtt-broker-server.ts

```diff
@@ -319,6 +232,110 @@ export default class MqttBrokerServer implements ProtocolServer {
+                                    `MqttBrokerServer at ${this.brokerURI}, packet has no Content Type and does not parse as JSON, relaying raw (string) payload.`
+                                );
+                                value = payload.toString();
+                            }
+                        }
+                    }
+                    thing
+                        .invokeAction(segments[3], value)
+                        .then((output) => {
+                            // MQTT cannot return results
+                            if (output) {
+                                console.warn(
+                                    `MqttBrokerServer at ${this.brokerURI} cannot return output '${segments[3]}'`
+                                );
+                            }
+                        })
+                        .catch((err) => {
+                            console.error(
+                                `MqttBrokerServer at ${this.brokerURI} got error on invoking '${segments[3]}': ${err.message}`
+                            );
+                        });
+
+                    // topic found and message processed
+                    return;
+                } // Action exists?
+            } // Thing exists?
+        } else if (segments.length === 5 && segments[4] === "writeproperty") {
+            // connecting to the writeable properties
+            const thing = this.things.get(segments[1]);
+            if (thing) {
+                if (segments[2] === "properties") {
+                    const property = thing.properties[segments[3]];
+                    if (property) {
+                        if (!property.readOnly) {
+                            thing.writeProperty(segments[3], JSON.parse(payload.toString())).catch((err) => {
+                                console.error(
+                                    "[binding-mqtt]",
+                                    `MqttBrokerServer at ${this.brokerURI} got error on writing to property '${segments[3]}': ${err.message}`
+                                );
+                            });
+                            // topic found and message processed
+                            return;
+                        } else {
+                            console.warn(
+                                "[binding-mqtt]",
+                                `MqttBrokerServer at ${this.brokerURI} received message for readOnly property at '${receivedTopic}'`
+                            );
+                            return;
+                        } // property is writeable? Not necessary since it didn't actually subscribe to this topic
+                    } // Property exists?
+                }
+            }
+            return;
+        }
+        // topic not found
+        console.warn(
+            "[binding-mqtt]",
+            `MqttBrokerServer at ${this.brokerURI} received message for invalid topic '${receivedTopic}'`
+        );
+    }
+
```

Comment:
Since we are moving this function, I would change also the API used with the `ExposedThing` interface. Now the code uses the old one, my preference is to use the new `handleX` functions.